agir/msgs/views.py

```diff
@@ -133,13 +98,21 @@ def get_queryset(self):
                 )
             )
             .distinct()
+            .order_by("-last_update", "-created")
         )
 
-        return sorted(
-            chain(private_group_messages, public_group_messages),
-            key=attrgetter("created", "last_update"),
-            reverse=True,
-        )
+        list_group_messages = list(group_messages)
+        # Remove messages where person is not in allowed membership types
+        for msg in list_group_messages:
+            if msg.author.id == person.id:
+                continue
+            user_permission = Membership.objects.get(
+                person_id=person.id, supportgroup_id=msg.supportgroup.id
+            ).membership_type
+            if msg.required_membership_type > user_permission:
+                list_group_messages.remove(msg)
+
+        return list_group_messages
```

Comment:
Pardon, ce n'était pas très clair. 
Ici tu pourrais faire : 
```suggestion
        return [ msg for msg in group_messages if person.role.has_perm("msgs.view_message", msg)
```
qui a l'avantage de centraliser toute la logique de qui a le droit de voir un message au même endroit, dans le fichier `rules`

Reply:
Ah oui en effet c'est plus centralisé ! C'est plus clair :)